banking/ebics/utils.py

```diff
@@ -234,57 +235,47 @@ def _create_bank_transaction(
 
 	https://www.joonis.de/en/fintech/doc/sepa/#fintech.sepa.SEPATransaction
 	"""
-	# sepa_transaction.bank_reference can be None, but we can still find an ID in the XML
-	# For our test bank, the latter is a timestamp with nanosecond accuracy.
-	transaction_id = (
-		sepa_transaction.bank_reference
-		or sepa_transaction._xmlobj.Refs.TxId.text
-		or get_transaction_hash(sepa_transaction)
-	)
-
-	# NOTE: This does not work for old data, this ID is different from Kosma's
-	if transaction_id and frappe.db.exists(
-		"Bank Transaction",
-		{"transaction_id": transaction_id, "bank_account": bank_account},
-	):
-		return
-
 	if start_date and sepa_transaction.date < start_date:
 		return
 
-	bt = frappe.new_doc("Bank Transaction")
-	bt.date = sepa_transaction.date
-	bt.bank_account = bank_account
-	bt.company = company
+	values_to_hash = [
+		sepa_transaction.date,
+		sepa_transaction.iban,
+		sepa_transaction.name,
+		sepa_transaction.eref,
+		sepa_transaction.amount.value,
+		sepa_transaction.amount.currency,
+		sepa_transaction.info,
+		*sepa_transaction.purpose,
+	]
 
 	amount = float(sepa_transaction.amount.value)
-	bt.deposit = max(amount, 0)
-	bt.withdrawal = abs(min(amount, 0))
-	bt.currency = sepa_transaction.amount.currency
-
-	bt.description = "\n".join(sepa_transaction.purpose) or sepa_transaction.info
-	bt.reference_number = sepa_transaction.eref
-	bt.transaction_id = transaction_id
-	bt.bank_party_iban = sepa_transaction.iban
-	bt.bank_party_name = sepa_transaction.name
-
-	with contextlib.suppress(frappe.exceptions.UniqueValidationError):
-		bt.insert()
-		bt.submit()
+	create_bank_transaction(
+		bank_account=bank_account,
+		transaction_id=(
+			# sepa_transaction.bank_reference can be None, but we can still find an ID in the XML
+			# For our test bank, the latter is a timestamp with nanosecond accuracy.
+			# sepa_transaction.bank_reference
+			# or sepa_transaction._xmlobj.Refs.TxId.text
+			# or
```

Comment:
Here I also made a stupid mistake by leaving this part commented out. 🙈 Changing the way `transaction_id` is determined makes duplicate detection impossible.